web/src/components/ErrorScreen.js

```diff
@@ -1,122 +1,141 @@
 import React from 'react';
 import Guild from './Guild';
 import {UserAvatar} from './DiscordIcon';
+import {CSSTransitionGroup} from 'react-transition-group';
 
-export default class ErrorScreen extends React.Component {
-  render() {
-    const props = this.makeInner();
-    if (!props) return null;
-
-    return (
+function ErrorScreen({header, footer, children}) {
+  return (
+    <div>
       <div className="front flex-vertical flex-spacer">
         <div className="application-icon">
           <div className="application-icon-inner" />
         </div>
         <div className="front-inner">
           <header>
-            <h1>{props.header}</h1>
+            <h1>{header}</h1>
           </header>
           <div className="scroller-wrap">
-            <div className="details scroller">{props.body}</div>
+            <div className="details scroller">{children}</div>
           </div>
-          <footer>{props.footer}</footer>
+          <footer>{footer}</footer>
         </div>
       </div>
+    </div>
+  );
+}
+
+export default class ErrorScreenWrapper extends React.Component {
+  renderNoAuth() {
+    return (
+      <ErrorScreen header="Huh?!">
+        <section>
+          <p>Well, this is embarrassing. I don't know how you got here!</p>
+          <p>This isn't normal at all, so I'm not really sure how to proceed...</p>
+          <p>Maybe try refreshing? Let's hope for the best!</p>
+        </section>
+      </ErrorScreen>
+    );
+  }
+
+  renderNoConnection() {
+    return (
+      <ErrorScreen header="I fell asleep!">
+        <section>
+          <p>Give me a couple of seconds to wake up, then try refreshing the page!</p>
+        </section>
+      </ErrorScreen>
+    );
+  }
+
+  renderNoAccess() {
+    const footer = (
+      <form method="POST" action="logout">
+        <button className="primary" type="submit">
+          Re-authenticate
+        </button>
+      </form>
+    );
+    return (
+      <ErrorScreen header="Access denied!" footer>

```

Comment:
are you absolutely certain this is what we want?

Reply:
Well, it didn't like footer = {footer} and {footer} so I dunno